hub/core/tests/test_storage_provider.py

```diff
@@ -34,18 +40,197 @@ def test_provider(provider=MappedProvider()):
     with pytest.raises(KeyError):
         provider[FILE_1]
 
+    provider.flush()
+
 
 def test_memory_provider():
-    test_provider(MemoryProvider("abc/def"))
+    check_storage_provider(memory_provider)
 
 
 def test_local_provider():
-    test_provider(LocalProvider("./hub_storage_local_test"))
+    check_storage_provider(local_provider)
 
 
 @pytest.mark.skipif(not s3_creds_exist(), reason="requires s3 credentials")
 def test_s3_provider():
-    test_provider(S3Provider("snark-test/hub_storage_s3_test"))
+    check_storage_provider(s3_provider)
+
+
+def test_lru_mem_local(benchmark):
```

Comment:
You're taking the benchmark fixture as a parameter but not using it. I'm not sure if pytest-benchmark will skip this benchmark or time the whole thing, but either way I'm not sure it's what we want. In the former case, remove the param. In the latter, we're going to include the `check_storage_provider` assertions and all the pytest overhead.

Reply:
https://pytest-benchmark.readthedocs.io/en/stable/

Reply:
Ah, my bad, left the argument accidentally while copy pasting tests. Removed.